SignalServiceKit/src/Util/OWSSwiftUtils.swift

```diff
@@ -15,8 +15,41 @@ public func assertOnQueue(_ queue: DispatchQueue) {
     }
 }
 
-public func owsFail(_ message: String) {
+// Once we're on Swift4.2 we can mark this as inlineable
+// @inlinable
+public func owsNotImplemented(file: String = #file,
+                              function: String = #function,
+                              line: Int = #line) {
+    owsProdFail("\(function) is not implemented.", file: file, function: function, line: line)
+}
+
+public func owsProdFail(_ message: String) {
     Logger.error(message)
     Logger.flush()
     assertionFailure(message)
 }
+
+// Once we're on Swift4.2 we can mark this as inlineable
+// @inlinable
+public func owsFail(_ rawMessage: String,
+                    file: String = #file,
+                    function: String = #function,
+                    line: Int = #line) {
+    let message = "\(file) \(function) \(line): \(rawMessage)"
+    Logger.error(message)
+    Logger.flush()
+    assertionFailure(message)
+}
+
+// Once we're on Swift4.2 we can mark this as inlineable
+// @inlinable
+public func owsProdFail(_ rawMessage: String,
```

Comment:
Presumably the naming of these methods will align with our decisions for the Obj-C macros.